romancal/ramp_fitting/ramp_fit_step.py

```diff
@@ -225,12 +225,9 @@ def ols_cas22(self, input_model, readnoise_model, gain_model):
         err = np.sqrt(var_poisson + var_rnoise)
         dq = output.dq.astype(np.uint32)
 
-        ramp_dq = np.zeros(dq.shape[1:], dtype=np.uint32)
-        # Turn the jump detection DQ into a ramp DQ
-        for i in range(dq.shape[1]):
-            for j in range(dq.shape[2]):
-                if np.any(dq[:, i, j] == dqflags.group["JUMP_DET"]):
-                    ramp_dq[i, j] = dqflags.group["JUMP_DET"]
+        # Propagate DQ flags forward.
+        ramp_dq = get_pixeldq_flags(
+            dq, input_model.pixeldq, slopes, err, gain)
```

Comment:
`stcal` updates the dq flags in place meaning any flags passed in from the input model should persist through to the output of the model. So there should be no need to take care of the existing flags.

Reply:
Never mind, I see these are the pixel dq not the group dq flags.